src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2130,12 +2130,29 @@ BuildJobTreeTaskList(Job *jobTree, PlannerRestrictionContext *plannerRestriction
 		List *assignedSqlTaskList = AssignTaskList(sqlTaskList);
 		AssignDataFetchDependencies(assignedSqlTaskList);
 
-		/* now assign merge task's data fetch dependencies */
+		/* if the parameters has not been resolved, record it */
+		job->parametersInJobQueryResolved =
+			!HasUnresolvedExternParamsWalker((Node *) job->jobQuery, NULL);
+
+		/*
+		 * Make final adjustments for the assigned tasks.
+		 *
+		 * First, update SELECT tasks' parameters resolved field.
+		 *
+		 * Second, assign merge task's data fetch dependencies.
+		 */
 		foreach(assignedSqlTaskCell, assignedSqlTaskList)
 		{
 			Task *assignedSqlTask = (Task *) lfirst(assignedSqlTaskCell);
-			List *assignedMergeTaskList = FindDependentMergeTaskList(assignedSqlTask);
 
+			/* we don't support parameters in the physical planner */
+			if (assignedSqlTask->taskType == SELECT_TASK)
+			{
```

Comment:
In my first iteration, I simply set `assignedSqlTask->parametersInQueryStringResolved ` to `false` as I  thought physical planner cannot handle parameters and we never get here with parameters. However, it turns  out that with CURSORs, we can get here with parameters (tough we fail when executing it).

Instead, I decided to make it more robust to any further changes and call `HasUnresolvedExternParamsWalker ` on the job query